src/Symfony/Component/Security/Core/Authentication/RememberMe/CacheTokenVerifier.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authentication\RememberMe;
+
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+ * @author Jordi Boggiano <j.boggiano@seld.be>
+ */
+class CacheTokenVerifier implements TokenVerifierInterface
```

Comment:
We probably need a dedicated "system" cache pool to wire this one as a service.
Here is an example we could take inspiration from: https://github.com/symfony/symfony/blob/1a87c72c1f67e909a59c8dbe5ba3cce71a5bbeea/src/Symfony/Bundle/SecurityBundle/Resources/config/security.php#L278-L282
And the conditional removal of the service for the case where cache is not available:
https://github.com/symfony/symfony/blob/fd02ec075d68d54ff69e948d021223234d70e367/src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/AddExpressionLanguageProvidersPass.php#L37-L39